source/material_model/drucker_prager.cc

```diff
@@ -29,56 +31,116 @@ namespace aspect
+                  bool_min_strain_rate = true;
+                }
+
+              // plasticity
+              const double sin_phi = std::sin(angle_of_internal_friction * degree_to_rad);
+              const double cos_phi = std::cos(angle_of_internal_friction * degree_to_rad);
+              const double strain_rate_effective_inv = 1/strain_rate_effective;
+              const double strength_inv_part = 6.0/(sqrt3*(3.0-sin_phi));
+              const double strength = (dim == 2 ?
+                                       pressure * sin_phi + cohesion * cos_phi
+                                       :
+                                       (cohesion*cos_phi*strength_inv_part)
+                                       + (sin_phi * strength_inv_part) * pressure);
+              const double eta_plastic     = 0.5 * strength * strain_rate_effective_inv;
+
+
+              const double composition_viscosity = std::max(std::min(eta_plastic, maximum_viscosity), minimum_viscosity);
+
+              Assert(dealii::numbers::is_finite(composition_viscosity),ExcMessage ("Error: Viscosity is not finite."));
+              if (/*this->get_parameters().newton_theta != 0 &&*/ derivatives != NULL)
+                {
+                  if (!bool_min_strain_rate && composition_viscosity < maximum_viscosity && composition_viscosity > minimum_viscosity)
+                    {
+                      composition_viscosity_strain_rate_derivatives = 0.5 * (eta_plastic * (-1/(edot_ii_strict * edot_ii_strict))) * strain_rate_deviator;
+                      composition_viscosity_pressure_derivatives = (dim == 2 ?
+                                                                    0.5 * sin_phi * strain_rate_effective_inv
+                                                                    :
+                                                                    (sin_phi*strength_inv_part));
+                    }
+                  else
+                    {
+                      composition_viscosity_strain_rate_derivatives = 0;
+                    }
+                }
+
+              out.viscosities[i] = composition_viscosity;
+              Assert(dealii::numbers::is_finite(out.viscosities[i]),ExcMessage ("Error: Averaged viscosity is not finite."));
+
+              if (/*this->get_parameters().newton_theta != 0 &&*/ derivatives != NULL)
+                {
+
+                  derivatives->viscosity_derivative_wrt_strain_rate[i] = composition_viscosity_strain_rate_derivatives;//Utilities::derivatives_weighed_p_norm_average(out.viscosities[i],volume_fractions, composition_viscosities, composition_viscosities_strain_rate_derivatives, viscosity_averaging_p);
+
+                  derivatives->viscosity_derivative_wrt_pressure[i] = composition_viscosity_pressure_derivatives;//Utilities::derivatives_weighed_p_norm_average(out.viscosities[i],volume_fractions, composition_viscosities, composition_viscosities_pressure_derivatives, viscosity_averaging_p);
+
+#ifdef DEBUG
```

Comment:
no need for this -- in release mode, the `Assert` statements will be replaced by empty statements (=nothing), and then the compiler will recognize that the double loop does nothing and remove it as well.